internal/name/safeconcat.go

```diff
@@ -0,0 +1,25 @@
+package name
```

Comment:
What about naming it `names` instead? It will avoid the collision between the import name and frequent `name` variables, in addition to follow the convention from the stdlib (`strings`, `maps`, `slices`).

Reply:
Sorry, I just realized the package already existed before 🙃

Reply:
The package exists already, but I like the idea.